great_expectations/execution_engine/sqlalchemy_execution_engine.py

```
@@ -1032,6 +1004,175 @@ def close(self) -> None:
+
+            domain_id = compute_domain_kwargs.to_id()
+            if domain_id not in queries:
+                queries[domain_id] = {
+                    "select": [],
+                    "metric_ids": [],
+                    "domain_kwargs": compute_domain_kwargs,
+                }
+
+            if self.engine.dialect.name == "clickhouse":
+                queries[domain_id]["select"].append(
+                    metric_fn.label(
+                        metric_to_resolve.metric_name.join(
+                            random.choices(string.ascii_lowercase, k=4)
+                        )
+                    )
+                )
+                queries[domain_id]["metric_ids"].append(metric_to_resolve.id)
+            else:
+                queries[domain_id]["select"].append(metric_fn.label(metric_to_resolve.metric_name))
+                queries[domain_id]["metric_ids"].append(metric_to_resolve.id)
+        return queries
+
+    def _count_query_parameters(self, selectable: sqlalchemy.Selectable, select_list: list) -> int:
+        """Count the total number of parameters in a query with the given select expressions.
+
+        Args:
+            selectable: The base selectable object
+            select_list: List of SELECT expressions to include in the query
+
+        Returns:
+            Total number of parameters that would be generated when the query is compiled
+        """
+        AVERAGE_PARAMS_PER_SELECT = 2
+        if isinstance(selectable, sqlalchemy.TextClause):
+            test_query = sa.select(*select_list).select_from(selectable.columns().subquery())
+        elif isinstance(selectable, (sqlalchemy.Select, sqlalchemy.TextualSelect)):
+            test_query = sa.select(*select_list).select_from(selectable.subquery())
+        elif isinstance(selectable, sa.sql.FromClause):
+            test_query = sa.select(*select_list).select_from(selectable)
+        else:
+            return len(select_list) * AVERAGE_PARAMS_PER_SELECT
```

In which cases is this true?

---

If the `selectable` instance is none of the types defined in the if statement. I didn't want to raise an exception and instead use the fallback.